group/asg/variables.tf

```diff
@@ -20,6 +20,10 @@ variable "lc_id" {
 }
 
 ## ASG parameters
+variable "asg_name_prefix" {
```

Comment:
I would make it clear what this is and call it `asg_name_override`. Default would need to be empty string to work with the conditional I suggested above.